### DISCUSSION TITLE
Next steps after installing Client code

### DISCUSSION THREAD

**Discussion by VesuviusGroup** [mozilla/hubs#5699] [Hubs Cloud Questions] [answered]
I have Hubs running in the cloud, and now want to develop some local client code. 
- I have the client setup and running locally. I run in the following problems:
- When I click on Admin, that page does not load. 
- When I click on one of my favorited rooms, it links to my aws hubs, not the localhost. 
- When I get the room id from a room on my cloud install and make url like this: https://localhost:8080/hub.html?hub_id=o9a2Qzs
It fails to load.
- Creating a new room gives me a "The Scene failed to load" message.  
It feels like there is a step missing from the client code docs.
How do I load a scene locally, so I can actually test code?

**Comment by pattersonbl2** [comment]
The steps in this repo may help you. 
https://github.com/mozilla/reticulum

These would be the backend services for hubs.

**Reply by VesuviusGroup** [reply]
But isn't the point that when I have connected my local client with the aws backend, so I can test my clientcode against the server.

**Reply by hrithikwins** [reply]
- For admin you have to go to admin folder and install the packages and run locally.. admin is a whole different project folder

**Reply by hrithikwins** [reply]
- I have some guesses here.... you need connect to personal cloud (`npm run login`) while in local development ... so it's linked to the rectilium server of mozilla hubs.. you might have to setup your own backend server (as the link shared by @Pattersonbl2 describes )  to link the same id (like your aws generated id to local id)..

**Reply by VesuviusGroup** [reply]
I found what the missing step is. The local client tried to load assests through the hubs.local:8080 domain. Npm run start has no services running on that Domain. You **HAVE** to add "**127.0.0.1   hubs.local**" to your hosts file. 

Secondly you should add localhost and hubs.local to the cors list. how to here: https://github.com/mozilla/hubs/discussions/3907

**Comment by VesuviusGroup** [comment, marked as answer]
I found what the missing step is. The local client tried to load assets through the hubs.local:8080 domain. Npm run start has no services running on that Domain. You **HAVE** to add "**127.0.0.1   hubs.local**" to your hosts file. 

Secondly you should add localhost and hubs.local to the cors list. how to here: https://github.com/mozilla/hubs/discussions/3907

None of this is documented on https://hubs.mozilla.com/docs/hubs-cloud-custom-clients.html#run-your-custom-client
